proof-systems/src/darlin/tests/final_darlin.rs

```diff
@@ -369,6 +370,15 @@ pub fn generate_test_data<'a, G1: AffineCurve, G2: AffineCurve, D: Digest + 'a,
         info.clone()
     ).unwrap();
 
+    let domain_h_size = std::cmp::max(index_vk.index_info.num_constraints, index_vk.index_info.num_variables).next_power_of_two();
+    let domain_k_size = index_vk.index_info.num_non_zero.next_power_of_two();
+    let segment_size = (segment_size - 1).next_power_of_two();
```

Comment:
Why segment_size-1? I'd expect the segment_size to be the size, and not the supported degree.